File: covid-19-tracking-app/src/types/covidReportDetail.ts
import { Covid19DataByDayRegion } from "./covid19DataByDayRegion";
import { Covid19DataByRegion, TodayReport } from "./covid19DataByRegion";
import { VaccineData } from "./vaccineData";

export type CovidReportDetail = {
    covidReport: Covid19DataByRegion,
    vaccineReport: VaccineData,
    covidReportByDay: Covid19DataByDayRegion[],
    today: TodayReport
};

File: COVID-19-Tracking-API/Domain/Models/Entities/CovidReportDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace C19Tracking.Domain.Models.Entities
{
    public class CovidReportDetail
    {
        public CovidDataByRegion CovidReport { get; set; }
        public List<CovidDataByDayRegion> CovidReportByDay { get; set; } 
        public VaccineData VaccineReport { get; set; } 
    }
}
